libs/auth/test/utils.ts

```diff
@@ -12,6 +18,25 @@ export function numericArray(input: {
   return Array.from<number>({ length: input.length }).fill(input.value ?? 0);
 }
 
+/**
+ * A mock card reader
+ */
+export class MockCardReader implements Pick<CardReader, 'transmit'> {
+  private readonly onReaderStatusChange: OnReaderStatusChange;
+
+  constructor(...params: ConstructorParameters<typeof CardReader>) {
+    this.onReaderStatusChange = params[0].onReaderStatusChange;
+  }
+
```

Comment:
TIL `ConstructorParameters`. I would prefer this because it makes it a tiny bit easier to read and consistent with your `CardReader` constructor signature, but this is a nitty nit nit.

```suggestion
  constructor(input: ConstructorParameters<typeof CardReader>[0]) {
    this.onReaderStatusChange = input.onReaderStatusChange;
  }
```

Reply:
Agree that what you have is more intuitive! It does always take me a second to process `...params`